### PR TITLE
Orientation arbiter

### PR DESCRIPTION
**Pull request by nathanestill**
Added a message, topic, and method to accept a position for the drone to look at.
The commit also has the QR stuff that I finished at the end of last semester that I don't quite know how to get rid of but is done anyway.

### CODE REVIEW

File: iarc_arbiter/scripts/transformers.py [hunk 1]
```diff
@@ -211,18 +211,19 @@ def cmd_pos(self, msg):
...
+            # Turn drone toward the provided orientation
+            _, _, angle_err = tf.transformations.euler_from_quaternion(
+                [getattr(pose.pose.orientation, s) for s in 'xyzw'])
+            angle_err += self.config.angle_offset
+
+            # Normalize the angle
+            while angle_err <= -np.pi:
+                angle_err += 2 * np.pi
+            while angle_err > np.pi:
+                angle_err -= 2 * np.pi
```
Comment (HALtheWise) [inline iarc_arbiter/scripts/transformers.py]: This angle normalization should probably happen even if the error was passed as an argument.

File: iarc_arbiter/scripts/transformers.py [hunk 2]
```diff
@@ -211,18 +211,19 @@ def cmd_pos(self, msg):
...
+        else:
+            angle_err = angle_error
```
Comment (HALtheWise) [inline iarc_arbiter/scripts/transformers.py]: This is a little awkward, perhaps just use angle_err as the passed argument and only overwrite it if it was None

File: iarc_arbiter/scripts/transformers.py [hunk 3]
```diff
@@ -270,6 +271,31 @@ def transformPoseFull(self, target_frame, ps, stationary):
         result.pose = Pose(Point(*xyz), Quaternion(*quat))
         return result
 
+class PIDPosCamController(object):
+    def __init__(self, tfl, ddynrec, pos_controller):
+        """
+        :type tfl: TransformListener
+        :type ddynrec: DDynamicReconfigure
+        :type alt_controller: PIDAltController
```
Comment (HALtheWise) [inline iarc_arbiter/scripts/transformers.py]: Should be `:type pos_controller: PIDPosController`

File: iarc_forebrain/scripts/util/Drone.py [hunk 4]
```diff
@@ -282,3 +283,40 @@ def _on_navdata(self, msg):
...
+        camMsg.look_at_position.pose.position.z = 0.0
+
+        camMsg.pose_stamped.pose.position.x = des_x
+        camMsg.pose_stamped.pose.position.y = des_y
+        camMsg.pose_stamped.pose.position.z = height
+        camMsg.pose_stamped.header.frame_id = frame
+        camMsg.look_at_position.header.frame_id = frame
+        self.camPosPub.publish(camMsg)
+
+    def turn_to(self, orientationQuat, frame = 'map'):
```
Comment (HALtheWise) [inline iarc_forebrain/scripts/util/Drone.py]: This should maybe take a simple angle in radians, and convert to a quaternion internally.

File: iarc_fuses/scripts/PicTesterPublisher.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python2
```
Comment (HALtheWise) [inline iarc_fuses/scripts/PicTesterPublisher.py]: It looks to me like there is some other stuff you worked on that hasn't been merged into master yet. We should figure out what branch that is and make sure it has a pull request.

**Comment by nathanestill** [comment]
I fixed the things you suggested, and we have it tested, so it should be good to merge

**Pull request merged**